contracts/NewForwarder.sol

```diff
@@ -0,0 +1,332 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.15;
+import '@openzeppelin/contracts/token/ERC1155/IERC1155.sol';
+import '@openzeppelin/contracts/token/ERC721/IERC721.sol';
+import '@openzeppelin/contracts/token/ERC721/IERC721Receiver.sol';
+import '@openzeppelin/contracts/token/ERC1155/utils/ERC1155Receiver.sol';
+import './ERC20Interface.sol';
+import './TransferHelper.sol';
+import './IForwarder.sol';
+
+/**
+ * Contract that will forward any incoming Ether to the parent address of the contract
+ *
+ */
+contract NewForwarder is IERC721Receiver, ERC1155Receiver, IForwarder {
+  // Address to which any funds sent to this contract will be forwarded
+  address public parentAddress;
+  address public feeAddress;
+  mapping(address => bool) public whiteListedAddresses;
```

Comment:
If we are not adding a function to update whiteListedAddresses with additional addresses, Do we need to add this mapping?, Just need to have a comparison in the modifier to check the address is  parentAddress or feeAddress, But if using mapping uses lesser gas, Its okay

Reply:
Modified this